### PR TITLE
DR-596 deploy service infrastructure workflow

### PR DESCRIPTION
**Pull request by jope1**
## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: scripts/action-infra-stack.sh [hunk 1]
```diff
@@ -53,8 +51,8 @@ else
   fi
 fi
 
-if [ -z "$TERRAFORM_WORKSPACE_NAME" ] ; then
-  echo Set TERRAFORM_WORKSPACE_NAME
+if [ -z "$WORKSPACE" ] ; then
```
Comment (timrickwood) [inline scripts/action-infra-stack.sh:54]: Is it worth updating the comments at top of script to be clear about the inputs etc

File: scripts/functions/git-functions.sh [hunk 2]
```diff
@@ -70,13 +70,13 @@ function check_commit_message_length {
 }
 
 # exports string to use as terraform workspace
-# equal to DEPLOYMENT_WORKSPACE if previously exported and not a R or V tag
+# equal to WORKSPACE if previously exported and not a R or V tag
 # or derived from branch name
 function export_terraform_workspace_name {
```
Comment (timrickwood) [inline scripts/functions/git-functions.sh:75]: Do we still need this function ?

File: .github/workflows/pipeline-deploy-app-develop.yaml [hunk 3]
```diff
@@ -10,33 +10,70 @@ jobs:
...
-      stack: '["github-runner", "terraform_management"]'
-      tf_version: ${{ needs.metadata.outputs.tf_max_version }}
+      environment: mgmt
+      workspace: ${{ needs.metadata.outputs.workspace }}
+      domain: "uec-dos-management"
+      stacks: "['test-stack-1', 'test-stack-2']"
+      action: plan
     secrets: inherit
 
+  deploy-infrastructure-apply:
```
Comment (timrickwood) [inline .github/workflows/pipeline-deploy-app-develop.yaml:43]: Are these for testing purposes

Comment (jope1) [reply .github/workflows/pipeline-deploy-app-develop.yaml:43]: Yep

File: scripts/action-infra-stack.sh (hunk 1 above)
Comment (jope1) [reply scripts/action-infra-stack.sh:52]: Done

File: .github/actions/action-infrastructure-stack/action.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+description: "Actions infrastructure stack of a given domain into a given environment and workspace."
+
+inputs:
+    environment:
+        description: "The name of the environment to action the infrastructure"
+        required: true
+    workspace:
+        description: "The name of the workspace to action the infrastructure into."
+        required: true
+    tag:
```
[the remark this answers is not in the document]
Comment (jope1) [reply .github/actions/action-infrastructure-stack/action.yaml]: Removed

**Review by timrickwood — APPROVED** [review]

**Review by stuartreid — APPROVED** [review]
LGTM

**Pull request closed without merging**